camply/notifications/telegram.py

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env python3
+
+# Author::    Justin Flannery  (mailto:juftin@juftin.com)
+
+"""
+Push Notifications via Telegram
+"""
+
+from abc import ABC
+from datetime import datetime
+import logging
+from typing import List, Optional
+
+import requests
+
+from camply.config import CampsiteContainerFields, TelegramConfig
+from .base_notifications import BaseNotifications
+from ..containers import AvailableCampsite
+
+logger = logging.getLogger(__name__)
+
+
+class TelegramNotifications(BaseNotifications, ABC):
+    """
+    Push Notifications via Telegram
+    """
+
+    def __init__(self):
+        if any([TelegramConfig.BOT_TOKEN is None,
+                TelegramConfig.BOT_TOKEN == "",
+                TelegramConfig.CHAT_ID is None,
+                TelegramConfig.CHAT_ID == ""]):
+            warning_message = (
+                "Telegram is not configured properly. To send Telegram messages "
+                "make sure to run `camply configure` or set the "
+                "proper environment variables: `TELEGRAM_BOT_TOKEN` and `TELEGRAM_CHAT_ID`.")
+            logger.error(warning_message)
+            raise EnvironmentError(warning_message)
+
+    def __repr__(self):
+        """
+        String Representation
+        """
+        return "<TelegramNotifications>"
+
+    @staticmethod
+    def send_message(message: str, escaped=False, **kwargs) -> Optional[requests.Response]:
+        """
+        Send a message via Telegram - if environment variables are configured
+
+        Parameters
+        ----------
+        message: str
+        escaped: bool
+
+        Returns
+        -------
+        Response
+        """
+        if not escaped:
+            message = TelegramNotifications.escape_text(message)
+
+        telegram_headers = TelegramConfig.API_HEADERS.copy()
+        message_json = TelegramConfig.API_CONTENT.copy()
+        message_json.update({"text": message})
+        logger.debug(message_json)
+        response = requests.post(url=TelegramConfig.API_ENDPOINT,
+                                 headers=telegram_headers,
+                                 json=message_json)
+        if response.status_code != 200:
```

Comment:
@stjohnjohnson one follow up question, would it be okay to raise an error here for a successful response?

Reply:
It's certainly okay to do that.  I was copying how it was done for PushBullet: https://github.com/juftin/camply/blob/main/camply/notifications/pushbullet.py#L68-L69

But it seems Pushover raises an error: https://github.com/juftin/camply/blob/main/camply/notifications/pushover.py#L69-L71

Reply:
Gotcha - good catch. I made a couple very small tweaks to the notifiers. Let me know if everything looks good and we'll publish